### DISCUSSION TITLE
[Bug?] Parent validation does not trigger all of embedded documents validation

### DISCUSSION THREAD

**Discussion by minhngocd** [mongodb/mongoid#5805] [Q&A]
Hello! We've just upgraded from v8.1.4 to v8.1.5 and our tests are failing, which alerted us to the following behaviour change.

We have a parent model with embedded list of child documents like so : 

```ruby
class Whole
  include Mongoid::Document
  
  embeds_many :parts
  accepts_nested_attributes_for :parts

  def errors
    parts.map do |part|
      part.errors.full_messages if part.errors.present?
    end
  end
end

class Part
  include Mongoid::Document

  embedded_in :whole
  
  field :title,      type: String
  validates :title, presence: true
end
```

Before the update, if I had a Whole with multiple failing invalid Parts, it would return all validation messages of all parts. Seems like previously .valid? was triggered on all children when .valid? was called on the parent (objects were created with nested attributes. 

```ruby
whole = Whole.new(parts: [{title: "1"}, {title: ""}, {title: ""}]

whole.valid? => false
whole.errors => ["Title can't be blank", "Title can't be blank"]
```

However AFTER the new update, we are no longer getting any validation error messages past the first one - unless the child part `.valid?` is explicitly called. 

```ruby
whole = Whole.new(parts: [{title: "1"}, {title: ""}, {title: ""}]

whole.valid? => false
whole.errors => ["Title can't be blank"]
whole.parts[1].errors => ["Title can't be blank"]
whole.parts[2].errors => []

whole.parts[2].valid? => false
whole.parts[2].errors => ["Title can't be blank"]
whole.errors => ["Title can't be blank"]

whole.valid? => false
whole.errors => ["Title can't be blank", "Title can't be blank"]
```

Additionally after some more debugging on the new version we found that the validations for multiple embedded association was NOT triggered when parts were created these ways:
- `whole.parts.build(....)`
- Nested attributes `Whole.new(..., parts: [{title:....}, {...}])`

But however DOES get triggered if you do
- `whole.parts << Part.new(...)` or
- `whole.parts = [Part.new(...), ...]`

Any light shed on this would be greatly appreciated. Thank you!

**Comment by jamis** [comment]
Thank you for reporting this. I've confirmed that it is behaving as you described, which does not seem right. I've created a ticket here (https://jira.mongodb.org/browse/MONGOID-5757) to track this.

**Comment by stanley90** [comment]
+1, but I found this while upgrading from 7.5.4 to 8.0.8

**Reply by jamis** [reply]
Good to know; I'll update the ticket with that extra info. 👍
